node/src/accountant/mod.rs

```diff
@@ -371,37 +366,31 @@ impl Handler<NodeFromUiMessage> for Accountant {
 }
 
 impl Accountant {
-    pub fn new(config: &mut BootstrapperConfig, dao_factories: DaoFactories) -> Accountant {
-        let payment_thresholds = Rc::new(
-            config
-                .payment_thresholds_opt
-                .take()
-                .expectv("Payment thresholds"),
-        );
-        let scan_intervals = config.scan_intervals_opt.take().expectv("Scan Intervals");
-        let when_pending_too_long_sec = config
-            .when_pending_too_long_opt
+    pub fn new(mut config: BootstrapperConfig, dao_factories: DaoFactories) -> Accountant {
```

Comment:
Please refer to this function `merge_unprivileged` and also consider discussing It with Dan. In my opinion I prefer to wrap it with an `Option`, because there are certain moments in time when the value is `None` and then suddenly it gets populated.

Reply:
I see your point and it's a good one. I need to add, "sadly", because it would be nice if we didn't have to use the options to initiate values sequentially (we do exactly that; leaving in a None until we get to a chance to create the values belonging with that place).

 I'm offering a shallow improvement (to be honest, I would go ahead and do it myself) by separating that port out of the level of filling the Accountant's new instance fields with values. Even though unhappy, I'd use a mutable reference and take care of the ever-supplied values inside another function like here:

    pub fn new(mut config: BootstrapperConfig, dao_factories: DaoFactories) -> Accountant {
        let (payment_thresholds, scan_intervals) = Self::unwrap_opt_fields(&mut config);
        let earning_wallet = Rc::new(config.earning_wallet);
        let financial_statistics = Rc::new(RefCell::new(FinancialStatistics::default()));

        Accountant {
        
Then the function itself would be:

    fn unwrap_opt_fields(config: &mut BootstrapperConfig) -> (PaymentThresholds, ScanIntervals) {
        let payment_thresholds = config
            .payment_thresholds_opt
            .take()
            .expectv("Payment thresholds");
        let scan_intervals = config.scan_intervals_opt.take().expectv("Scan Intervals");
        (payment_thresholds, scan_intervals)
    }

Reply:
Ok, I did that.